cluster/src/main/java/org/apache/iotdb/cluster/query/last/ClusterLastQueryExecutor.java

```diff
@@ -105,29 +109,39 @@ public ClusterLastQueryExecutor(LastQueryPlan lastQueryPlan, MetaGroupMember met
       results.add(new Pair<>(true, new TimeValuePair(Long.MIN_VALUE, null)));
     }
 
-    List<PartitionGroup> globalGroups = metaGroupMember.getPartitionTable().getGlobalGroups();
-    List<Future<List<Pair<Boolean, TimeValuePair>>>> groupFutures =
-        new ArrayList<>(globalGroups.size());
-    List<Integer> dataTypeOrdinals = new ArrayList<>(dataTypes.size());
-    for (TSDataType dataType : dataTypes) {
-      dataTypeOrdinals.add(dataType.ordinal());
+    Map<PartitionGroup, List<PartialPath>> groupToPaths = new HashMap<>();
+    Map<PartitionGroup, List<TSDataType>> groupToDataTypes = new HashMap<>();
+    try {
+      for (int i = 0; i < seriesPaths.size(); i++) {
+        PartialPath path = seriesPaths.get(i);
+        TSDataType dataType = dataTypes.get(i);
+        // TODO: doesn't work if time series is partitioned by time
+        PartitionGroup group = metaGroupMember.getPartitionTable().partitionByPathTime(path, 0);
```

Comment:
I think we should add a `if-else` to check whether timePartition is enabled:
- If it is enabled, it should pass the query to all data groups, as we don't know where the last value is in.
- If it is disabled, we can do an optimization that passing the query to the data group owing the timeseries.